xen_vm_disk.py

```diff
@@ -50,37 +56,56 @@ def get_diskstats_dict():
 
         x = line.strip().split()
         disk_name = x.pop(2)
-        diskstats_dict[disk_name]={}
+        diskstats_dict[disk_name] = {}
         for i in header:
             diskstats_dict[disk_name][i] = x.pop(0)
 
     return(diskstats_dict)
 
-def get_dmname_from_majorminor(diskstats=False, major=0,minor=0):
+
+def get_dmname_from_majorminor(diskstats=False, major=0, minor=0):
     for disk in diskstats:
-        if int(diskstats[disk]['minor']) == minor and int(diskstats[disk]['major']) == major:
+        if int(
+                diskstats[disk]['minor']) == minor and int(
+                diskstats[disk]['major']) == major:
             return(disk)
     return(False)
 
-graphite_data=''
-hostname = socket.gethostname().replace('.','_')
+graphite_data = ''
+hostname = socket.gethostname().replace('.', '_')
 now = str(int(time.time()))
-sector_size=512
-xenstore_vbd = subprocess.Popen("/usr/sbin/xenstore-ls ", shell=True, bufsize=8192, stdout=subprocess.PIPE).stdout.readlines()
-xmlist = subprocess.Popen("/usr/sbin/xm list -l", shell=True, bufsize=128000, stdout=subprocess.PIPE).stdout.readlines()
-
-diskstats=get_diskstats_dict()
-vservers=get_vbds()
+sector_size = 512
+xenstore_vbd = subprocess.Popen(
+    "/usr/sbin/xenstore-ls ",
+    shell=True,
+    bufsize=8192,
+    stdout=subprocess.PIPE).stdout.readlines()
+xmlist = subprocess.Popen(
+    "/usr/sbin/xm list -l",
+    shell=True,
+    bufsize=128000,
+    stdout=subprocess.PIPE).stdout.readlines()
+
+diskstats = get_diskstats_dict()
+vservers = get_vbds()
 
 for server in vservers:
     for device in vservers[server]:
-        major,minor = vservers[server][device].strip("'").split(':')
-        dmname=get_dmname_from_majorminor(diskstats,int(major,16),int(minor,16))
-
-        graphite_data += 'servers.%s.virtualisation.vserver.%s.disk.%s.bytesRead %s %s\n' % (hostname, server.replace('.','_'),  device, str(int(diskstats[dmname]['sec_read'])*sector_size), now )
-        graphite_data += 'servers.%s.virtualisation.vserver.%s.disk.%s.bytesWrite %s %s\n' % (hostname, server.replace('.','_'), device, str(int(diskstats[dmname]['sec_written'])*sector_size), now )
-        graphite_data += 'servers.%s.virtualisation.vserver.%s.disk.%s.iopsRead %s %s\n' % (hostname, server.replace('.','_'),   device, str(diskstats[dmname]['reads']), now )
-        graphite_data += 'servers.%s.virtualisation.vserver.%s.disk.%s.iopsWrite %s %s\n' % (hostname, server.replace('.','_'),  device, str(diskstats[dmname]['writes']), now )
-        graphite_data += 'servers.%s.virtualisation.vserver.%s.disk.%s.ioTimeMs %s %s\n' % (hostname, server.replace('.','_'),   device, str(diskstats[dmname]['ms_io']), now )
+        major, minor = vservers[server][device].strip("'").split(':')
+        dmname = get_dmname_from_majorminor(
+            diskstats, int(
+                major, 16), int(
+                minor, 16))
```

Comment:
`get_dmname_from_majorminor` wants a bool as first argument. Maybe we should get rid of the `False` as default in these functions in all `xen*` scripts.